File: slider/src/androidTest/java/com/adsamcik/slider/IntValueSliderTest.java
package com.adsamcik.slider;


import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.test.InstrumentationRegistry;

import junit.framework.Assert;

import org.junit.Test;

public class IntValueSliderTest {
	private final Context appContext = InstrumentationRegistry.getTargetContext();

	private final Integer[] ints = new Integer[]{0, 1, 2, 4, 7, 12};

	@Test
	public void sharedPreferences() throws Exception {
		final SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(appContext);
		final String prefName = "TESTING PREFERENCE";
		final int noPreference = -1;

		preferences.edit().remove(prefName).commit();

		IntValueSlider slider = new IntValueSlider(appContext);
		slider.setItems(ints);

		slider.setPreferences(preferences, prefName, 4);

		Assert.assertEquals(4L, (long) slider.getValue());

		slider.setValue(7);

		int value = slider.getValue();
		Assert.assertEquals(value, preferences.getInt(prefName, noPreference));

		slider.removePreferences();

		slider.setValue(1);

		Assert.assertEquals(value, preferences.getInt(prefName, noPreference));
		Assert.assertEquals((long) 1, (long) slider.getValue());

		//cleanup
		preferences.edit().remove(prefName).apply();
	}
}

File: slider/src/androidTest/java/com/adsamcik/slider/IntValueSliderTest.kt
package com.adsamcik.slider


import android.content.Context
import androidx.preference.PreferenceManager
import androidx.test.core.app.ApplicationProvider
import com.adsamcik.slider.extensions.IntSliderSharedPreferencesExtension
import com.adsamcik.slider.implementations.IntValueSlider
import org.junit.Assert
import org.junit.Test

class IntValueSliderTest {
	private val appContext = ApplicationProvider.getApplicationContext<Context>()

	private val ints = arrayOf(0, 1, 2, 4, 7, 12)

	@Test
	@Throws(Exception::class)
	fun sharedPreferences() {
		val preferences = PreferenceManager.getDefaultSharedPreferences(appContext)
		val prefName = "TESTING PREFERENCE"
		val noPreference = -1

		preferences.edit().clear().apply()

		val slider = IntValueSlider(appContext)
		slider.setItems(ints)

		val extension = IntSliderSharedPreferencesExtension(preferences, prefName, 4)
		slider.addExtension(extension)

		Assert.assertEquals(4L, slider.value.toLong())

		slider.value = 7

		val value = slider.value
		Assert.assertEquals(value, preferences.getInt(prefName, noPreference))

		slider.removeExtension(extension)

		slider.value = 1

		Assert.assertEquals(value, preferences.getInt(prefName, noPreference))
		Assert.assertEquals(1.toLong(), slider.value.toLong())

		//cleanup
		preferences.edit().remove(prefName).apply()
	}
}
